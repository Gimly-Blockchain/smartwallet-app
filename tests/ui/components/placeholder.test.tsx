import React from 'react'
import { shallow } from 'enzyme'
import Placeholder from '../../../src/ui/recovery/seedPhrase/components/placeholder'

describe('placeholder component', () => {
  const makeProps = (otherProps: {}) => ({
    i: 0,
    sorting: {},
    onPress: jest.fn(),
    ...otherProps,
  })

  it('matches the snapshot without sorting', () => {
    const props = makeProps({})
    const rendered = shallow(<Placeholder {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with sorting without being used', () => {
    const props = makeProps({ sorting: { 1: 'Seed', 5: 'Phrase' } })
    const rendered = shallow(<Placeholder {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with sorting with being used', () => {
    const props = makeProps({ sorting: { 0: 'Seed', 5: 'Phrase' } })
    const rendered = shallow(<Placeholder {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
