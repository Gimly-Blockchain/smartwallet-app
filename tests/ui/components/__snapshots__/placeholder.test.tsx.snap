// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder component matches the snapshot with sorting with being used 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#942f51",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "width": 109,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffefdf",
        "fontFamily": "TT Commons",
        "fontSize": 20,
        "lineHeight": 26,
        "textAlign": "center",
      }
    }
  >
    Seed
  </Text>
</TouchableOpacity>
`;

exports[`placeholder component matches the snapshot with sorting without being used 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#942f51",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "width": 109,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffefdf",
        "fontFamily": "TT Commons",
        "fontSize": 20,
        "lineHeight": 26,
        "textAlign": "center",
      }
    }
  >
    1.
  </Text>
</TouchableOpacity>
`;

exports[`placeholder component matches the snapshot without sorting 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#942f51",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "width": 109,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffefdf",
        "fontFamily": "TT Commons",
        "fontSize": 20,
        "lineHeight": 26,
        "textAlign": "center",
      }
    }
  >
    1.
  </Text>
</TouchableOpacity>
`;

exports[`seedPhrase component matches the snapshot with sorting with being used 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#942f51",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "width": 109,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffefdf",
        "fontFamily": "TT Commons",
        "fontSize": 20,
        "lineHeight": 26,
        "textAlign": "center",
      }
    }
  >
    Seed
  </Text>
</TouchableOpacity>
`;

exports[`seedPhrase component matches the snapshot with sorting without being used 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#942f51",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "width": 109,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffefdf",
        "fontFamily": "TT Commons",
        "fontSize": 20,
        "lineHeight": 26,
        "textAlign": "center",
      }
    }
  >
    1.
  </Text>
</TouchableOpacity>
`;

exports[`seedPhrase component matches the snapshot without sorting 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#942f51",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 32,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "width": 109,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffefdf",
        "fontFamily": "TT Commons",
        "fontSize": 20,
        "lineHeight": 26,
        "textAlign": "center",
      }
    }
  >
    1.
  </Text>
</TouchableOpacity>
`;
